### PR TITLE
Add bank order filter

### LINKED ISSUE #1052: Need to adopt new ADC/TDC bank reader (for bg-merging, denoising, etc)

### PR DESCRIPTION
**Pull request by baltzell**
closes #1052

### CODE REVIEW

File: common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawOrderType.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * These order categories are all mutually exclusive, and denoising is
+ * categorized into levels, from 0 (least noisy) to 4 (most noisy).
+ * 
+ * @author baltzell
+ */
+public enum RawOrderType {
+
+    NOISE0          (  0),  // normal hits retained by denoising level-0
+    BGADDED         ( 10),  // hits added by background merging
```
Comment (raffaelladevita) [inline common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawOrderType.java:17]: Just for clarity: should 10 be labeled as "hits added by background merging and retained by denoising"?

File: common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawDataBank.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+     */
+    public int component(int index){
+        return this.intValue("component", index);
+    }
+    
+    /**
+     * @param index filtered index
+     * @return adc value
+     */
+    public int adc(int index){
```
Comment (raffaelladevita) [inline common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawDataBank.java:67]: adc banks have additional variables, I assume we will need methods for those as well?

Comment (baltzell) [reply common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawDataBank.java:67]: no

File: common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawOrderType.java (hunk 1 above)
Comment (baltzell) [reply common-tools/clas-detector/src/main/java/org/jlab/detector/banks/RawOrderType.java:17]: no, that is in Gagik's granite, I was just trying to get the ball rolling here

**Pull request merged**